ruby/neo4j/driver/internal/util/error_util.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+module Neo4j::Driver::Internal::Util
+  class ErrorUtil
+    DEFAULT_CONN_TERMINATED_REASON = 'Please ensure that your database is listening on the correct host and port and '\
+      'that you have compatible encryption settings both on Neo4j server and driver. '\
+      'Note that the default encryption setting has changed in Neo4j 4.0.'.freeze
+
+    SEC_EXCEPTION_CODE_MAPPING = {
+      'neo.clienterror.security.unauthorized': Neo4j::Driver::Exceptions::AuthenticationException,
+      'neo.clienterror.security.authorizationexpired': Neo4j::Driver::Exceptions::AuthorizationExpiredException,
+      'neo.clienterror.security.tokenexpired': Neo4j::Driver::Exceptions::TokenExpiredException
```

Comment:
do not downcase

Reply:
I see the use of `equalsIgnoreCase`, this gave me the impression that I shouldn't rely on error code's case. hence the down casing and comparing.
https://github.com/neo4j/neo4j-java-driver/blob/5.0/driver/src/main/java/org/neo4j/driver/internal/util/ErrorUtil.java#L74

Reply:
Yes, that's surprising. I think java code is inconsistent on that. In any case, do not change literals.